src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs

```diff
@@ -0,0 +1,4224 @@
+    {
+        Func<CancellationToken, Task<bool>> f = async (CancellationToken token) =>
+        {
+            await MethodAsync(token);
+            return true;
+        };
+        f(ct);
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c) => Task.CompletedTask;
+}
+            ";
+            return VerifyCS.VerifyCodeFixAsync(originalCode, fixedCode);
+        }
+
+        [Fact]
+        public Task CS_Diagnostic_FuncDelegateAwait_WithConfigureAwait()
+        {
+            string originalCode = @"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        Func<CancellationToken, Task<bool>> f = async (CancellationToken token) =>
+        {
+            await [|MethodAsync|]().ConfigureAwait(true);
+            return true;
+        };
+        f(ct);
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c) => Task.CompletedTask;
+}
+            ";
+            string fixedCode = @"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        Func<CancellationToken, Task<bool>> f = async (CancellationToken token) =>
+        {
+            await MethodAsync(token).ConfigureAwait(true);
+            return true;
+        };
+        f(ct);
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c) => Task.CompletedTask;
+}
+            ";
+            return VerifyCS.VerifyCodeFixAsync(originalCode, fixedCode);
+        }
+
+        [Fact]
+        public Task CS_Diagnostic_FuncDelegateAwaitOutside()
+        {
+            string originalCode = @"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        Func<CancellationToken, Task> f = (CancellationToken c) => [|MethodAsync|]();
+        await f(ct);
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c) => Task.CompletedTask;
+}
+            ";
+            string fixedCode = @"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        Func<CancellationToken, Task> f = (CancellationToken c) => MethodAsync(c);
+        await f(ct);
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c) => Task.CompletedTask;
+}
+            ";
+            return VerifyCS.VerifyCodeFixAsync(originalCode, fixedCode);
+        }
+
+        [Fact]
+        public Task CS_Diagnostic_NestedFunctionAwait()
+        {
+            string originalCode = @"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        async void LocalMethod(CancellationToken token)
+        {
+            await [|MethodAsync|]();
```

Comment:
Do we have a test where `LocalMethod` does not take a token, but outer method `M` has a token parameter? I presume this should generate a diagnostic, but no fix?

Reply:
Yep. But I will ignore this question as you requested below.😄